cli/internal/runcache/runcache.go

```diff
@@ -198,17 +198,32 @@ func (tc TaskCache) RestoreOutputs(ctx context.Context, prefixedUI *cli.Prefixed
 	case util.FullTaskOutput:
 		progressLogger.Debug("log file", "path", tc.LogFileName)
 		prefixedUI.Info(fmt.Sprintf("cache hit, replaying output %s", ui.Dim(tc.hash)))
-		if tc.LogFileName.FileExists() {
-
-			tc.rc.logReplayer(progressLogger, prefixedUI, tc.LogFileName)
-		}
+		tc.ReplayLogFile(prefixedUI, progressLogger)
+	case util.ErrorTaskOutput:
+		// The task succeeded, so we don't output anything in this case
 	default:
 		// NoLogs, do not output anything
 	}
 
 	return true, nil
 }
 
+// ReplayLogFile writes out the stored logfile to the terminal
+func (tc TaskCache) ReplayLogFile(prefixedUI *cli.PrefixedUi, progressLogger hclog.Logger) {
+	if tc.LogFileName.FileExists() {
+		tc.rc.logReplayer(progressLogger, prefixedUI, tc.LogFileName)
+	}
+}
+
+// OnError
+// If output-mode is set to "errors-only", we'll have buffered output
+// to disk without logging it.  Now we can log that here.
+func (tc TaskCache) OnError(terminal *cli.PrefixedUi, logger hclog.Logger) {
```

Comment:
This is annoying, but lint requires the code comment to be a whole line starting with `OnError`. You can still break it up into multiple lines, but I guess specifically what you have here is not acceptable 🤦🏾 ...